src/parsers/manifestjson.js

```diff
@@ -623,6 +627,36 @@ export default class ManifestJSONParser extends JSONParser {
     if (this.parsedJSON.homepage_url) {
       this.validateHomePageURL(this.parsedJSON.homepage_url);
     }
+
+    this.validateRestrictedPermissions();
+  }
+
+  validateRestrictedPermissions() {
+    const permissionsInManifest = (this.parsedJSON.permissions || []).map(
+      (permission) => String(permission).toLowerCase()
+    );
+
+    if (permissionsInManifest.length === 0) {
+      return;
+    }
+
+    const { firefoxMinVersion: minVersionSetInManifest } = this.getMetadata();
+
+    for (const permission of this.restrictedPermissions.keys()) {
+      if (permissionsInManifest.includes(permission)) {
+        const permMinVersion = this.restrictedPermissions.get(permission);
+
+        if (
+          !minVersionSetInManifest ||
+          mozCompare(String(minVersionSetInManifest), permMinVersion) === -1
```

Comment:
We should add explicit coverage for a non string min version value set in the manifest, which will make us sure that the linter will still complete successfully in that case (with the expected errors).

At the moment, it looks that removing `String(...)` here wouldn't make any of the existing tests to fail.